### PR TITLE
Task 9

### PR DESCRIPTION
**Pull request by alimovdavron**
All the tasks are implemented including optional ones.

3 - A working and correct express application should be in the bff-service folder. The reviewer can start this application locally with any valid configuration in the .env file and this application should works as described in task 9.1 

**http://alimovdavron-bff-api-develop.eu-central-1.elasticbeanstalk.com/**
[FE pull request
](https://github.com/alimovdavron/shop-react-redux-cloudfront/pull/6)

5 - The bff-service should be deployed with Elastic Beanstalk. The bff-service call should be redirected to the appropriate service : product-service or CART. The response from the bff-service should be the same as if product-service or CART services were called directly. 

**http://alimovdavron-bff-api-develop.eu-central-1.elasticbeanstalk.com/products**

+1 - Add a cache at the bff-service level for a request to the getProductsList function of the product-service. The cache should expire in 2 minutes. 

https://github.com/alimovdavron/node-aws-be/blob/task-9/bff-service/src/cache.interceptor.ts
https://github.com/alimovdavron/node-aws-be/blob/task-9/bff-service/src/cache.provider.ts

+1 - Use NestJS to create bff-service instead of express - **Done**

**UPD**
Redeployed using Node.js platform instead of Docker
image

### LINKS TO COMPARE BFF AND ORIGINAL 

**get products**
http://alimovdavron-bff-api-develop.eu-central-1.elasticbeanstalk.com/products
https://fw3b67j1u8.execute-api.eu-central-1.amazonaws.com/dev/products

**get product**
http://alimovdavron-bff-api-develop.eu-central-1.elasticbeanstalk.com/products/576bb6bf-6274-410c-89b4-a6a2cfb5cf60
https://fw3b67j1u8.execute-api.eu-central-1.amazonaws.com/dev/products/576bb6bf-6274-410c-89b4-a6a2cfb5cf60

**get product error**
https://fw3b67j1u8.execute-api.eu-central-1.amazonaws.com/dev/products/123
http://alimovdavron-bff-api-develop.eu-central-1.elasticbeanstalk.com/products/123

You can also test integraion on the Frontend - https://d1s4qelxm95ilt.cloudfront.net/

### CODE REVIEW

File: bff-service/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+WORKDIR /app
+
+COPY --from=build ./app/package*.json /app/
+RUN npm ci --only=production && npm cache clean --force
+
+COPY --from=build ./app/dist /app/
+
+EXPOSE 4000
+
+ENTRYPOINT ["node", "main"]
```
Comment (unfalse) [inline bff-service/Dockerfile]: Are we using docker in this task?

Comment (alimovdavron) [reply bff-service/Dockerfile]: Yep, my bad, will redeploy it today

**Comment by NikG77** [comment]
Не удается получить доступ к сайту Не удалось найти IP-адрес сервера alimovdavron-bff-service-dev.eu-central-1.elasticbeanstalk.com.

**Comment by alimovdavron** [comment]
> Не удается получить доступ к сайту Не удалось найти IP-адрес сервера alimovdavron-bff-service-dev.eu-central-1.elasticbeanstalk.com.



> Не удается получить доступ к сайту Не удалось найти IP-адрес сервера alimovdavron-bff-service-dev.eu-central-1.elasticbeanstalk.com.

Я обновил название сервиса в соответствии с требованиями и забыл поменять условие, сейчас поменяю

**Pull request merged**